packages/mask/src/plugins/EVM/UI/Web3State/getAssetsFn.ts

```diff
@@ -123,92 +119,6 @@ export const getFungibleAssetsFn =
 
         return uniqBy(
             [...nativeTokens, ...allTokens],
-            (x) => `${x.token.chainId}_${formatEthereumAddress(x.token.id)}`,
+            (x) => `${x.token.chainId}_${formatEthereumAddress(x.token.address)}`,
         ).sort(makeSortAssertWithoutChainFn())
     }
-
-export const getNonFungibleTokenFn =
-    (context: Web3ProviderType) =>
-    async (
-        address: string,
-        pagination: Pagination,
-        providerType?: string,
-        network?: Web3Plugin.NetworkDescriptor,
-    ): Promise<Pageable<Web3Plugin.NonFungibleToken>> => {
-        const socket = await context.providerSocket
-        let tokenInDb: ERC721TokenDetailed[] = []
-        // validate and show trusted erc721 token in first page
-        if (pagination?.page === 0) {
-            const trustedTokens = context.erc721Tokens.getCurrentValue()
-            const calls = trustedTokens.map(async (x) => {
-                const web3 = new Web3(
-                    createExternalProvider(
-                        context.request,
-                        () => ({
-                            ...context.getSendOverrides?.(),
-                            chainId: x.contractDetailed.chainId,
-                        }),
-                        context.getRequestOptions,
-                    ),
-                )
-                const contract = createContract<ERC721>(web3, x.contractDetailed.address, ERC721ABI as AbiItem[])
-                if (!contract) return null
-                const tokenDetailed = await getERC721TokenDetailedFromChain(x.contractDetailed, contract, x.tokenId)
-                const info = await getERC721TokenAssetFromChain(tokenDetailed?.info.tokenURI)
-                if (tokenDetailed && info)
-                    tokenDetailed.info = {
-                        ...info,
-                        ...tokenDetailed.info,
-                        hasTokenDetailed: true,
-                        name: info.name ?? tokenDetailed.info.name,
-                    }
-                return tokenDetailed
-            })
-
-            const fromChain = await Promise.all(calls)
-            tokenInDb = fromChain.filter(Boolean) as any[]
-        }
-
-        const socketId = `mask.fetchNonFungibleCollectibleAsset_${address}`
-        socket.send({
-            id: socketId,
-            method: 'mask.fetchNonFungibleCollectibleAsset',
-            params: { address, pageSize: 40 },
-        })
-
-        const tokenFromProvider = socket.getResult<ERC721TokenDetailed>(socketId)
```

Comment:
socket.getResult<ERC721TokenDetailed>(socketId) would not return any data until socket event param`info.done === true`, using `useSocket()` instead.